libraries/AP_Notify/LP5562.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+  LP5562 I2C driver
+*/
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* LED driver for LP5562 */
+
+#include "LP5562.h"
+
+#if AP_NOTIFY_LP5562_ENABLED
+
+#include <utility>
+
+#include <AP_HAL/AP_HAL.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define LP5562_LED_BRIGHT  255    // full brightness
+#define LP5562_LED_MEDIUM  170    // medium brightness
+#define LP5562_LED_DIM     85     // dim
+#define LP5562_LED_OFF     0      // off
+
+enum class Register {
+    ENABLE  = 0x00,
+    B_PWM   = 0x02,
+    G_PWM   = 0x03,
+    R_PWM   = 0x04,
+    CONFIG  = 0x08,
+    LED_MAP = 0x70,
+};
+
+LP5562::LP5562(uint8_t bus, uint8_t addr)
```

Comment:
maybe we should make addr 0 probe for all 4 possible addresses, 0x30, 0x31, 0x32 and 0x33

Reply:
We can add that later when someone asks for it :-)

Until then we'd just be wasting bytes.